server/services/store/sqlstore/board.go

```diff
@@ -0,0 +1,483 @@
+		"channel_id":          board.ChannelID,
+		"created_by":          board.CreatedBy,
+		"modified_by":         userID,
+		"type":                board.Type,
+		"title":               board.Title,
+		"description":         board.Description,
+		"icon":                board.Icon,
+		"show_description":    board.ShowDescription,
+		"is_template":         board.IsTemplate,
+		"properties":          propertiesBytes,
+		"card_properties":     cardPropertiesBytes,
+		"column_calculations": columnCalculationsBytes,
+		"create_at":           board.CreateAt,
+		"update_at":           board.UpdateAt,
+		"delete_at":           board.DeleteAt,
+	}
+
+	now := utils.GetMillis()
+
+	if existingBoard != nil {
+		query := s.getQueryBuilder(db).Update(s.tablePrefix+"boards").
+			Where(sq.Eq{"id": board.ID}).
+			Set("modified_by", userID).
+			Set("type", board.Type).
+			Set("title", board.Title).
+			Set("description", board.Description).
+			Set("icon", board.Icon).
+			Set("show_description", board.ShowDescription).
+			Set("is_template", board.IsTemplate).
+			Set("properties", propertiesBytes).
+			Set("card_properties", cardPropertiesBytes).
+			Set("column_calculations", columnCalculationsBytes).
+			Set("update_at", now).
+			Set("delete_at", board.DeleteAt)
+
+		if _, err := query.Exec(); err != nil {
+			s.logger.Error(`InsertBoard error occurred while updating existing board`, mlog.String("boardID", board.ID), mlog.Err(err))
+			return nil, err
+		}
+	} else {
+		insertQueryValues["created_by"] = userID
+		insertQueryValues["create_at"] = now
+		insertQueryValues["update_at"] = now
+
+		query := insertQuery.SetMap(insertQueryValues).Into(s.tablePrefix + "boards")
+		if _, err := query.Exec(); err != nil {
+			return nil, err
+		}
+	}
+
+	// writing board history
+	query := insertQuery.SetMap(insertQueryValues).Into(s.tablePrefix + "boards_history")
+	if _, err := query.Exec(); err != nil {
+		return nil, err
+	}
+
+	return s.getBoard(db, board.ID)
+}
+
+func (s *SQLStore) patchBoard(db sq.BaseRunner, boardID string, boardPatch *model.BoardPatch, userID string) (*model.Board, error) {
+	existingBoard, err := s.getBoard(db, boardID)
+	if err != nil {
+		return nil, err
+	}
+	if existingBoard == nil {
+		return nil, BoardNotFoundErr{boardID}
+	}
+
+	board := boardPatch.Patch(existingBoard)
+	return s.insertBoard(db, board, userID)
+}
+
+func (s *SQLStore) deleteBoard(db sq.BaseRunner, boardID, userID string) error {
+	now := utils.GetMillis()
+
+	board, err := s.getBoard(db, boardID)
+	if err != nil {
+		fmt.Printf("error on get board: %s\n", err)
+		return err
+	}
+
+	insertQuery := s.getQueryBuilder(db).Insert(s.tablePrefix+"boards_history").
+		Columns(
+			"team_id",
+			"id",
+			"type",
+			"modified_by",
+			"update_at",
+			"delete_at",
+		).
+		Values(
+			board.TeamID,
+			boardID,
+			board.Type,
+			userID,
+			now,
+			now,
+		)
+
+	if _, err := insertQuery.Exec(); err != nil {
+		return err
+	}
+
+	deleteQuery := s.getQueryBuilder(db).
```

Comment:
I know this is basically the same behaviour as workspaces, but should we think about what happens to the orphaned records in the `blocks` table?  This can be a follow-up discussion.